soroban-env-host/src/native_contract/token/contract.rs

```diff
@@ -297,6 +301,14 @@ impl TokenTrait for Token {
         Ok(())
     }
 
+    fn is_admin(e: &Host, address: Address) -> Result<bool, HostError> {
```

Comment:
Why do we need this and not `admin` function that returns an address?

Reply:
I've read the issue btw, but I don't think the argument there makes much sense, given that admin interface is separate from the common token interface (so multi-admin case would naturally require a different interface).

Reply:
Two reasons - 
1. It's possible for an admin to not exist. The lumen contract doesn't have one and other custom impls might do that same. We could return an option in those scenarios, but I thought this was simpler.
2. is_admin works for contracts that have multiple admins (idk if this really matters though).

Reply:
I think it would be correct for `admin` function to fail when used with XLM (the same way `set_admin` fails). For `is_admin`, I agree that returning false is correct. As for multiple admins, as I've mentioned above, I don't think we need a requirement for SAC admin interface to be compatible with multi-admin token interface, given that neither is a part of standard token interface.

I just feel like users might want `admin` anyway (e.g. for off-chain use cases) and `is_admin` seems really limiting. I guess we could have both as well...

Reply:
I think `admin` with an `Option<Address>` return value makes more sense. If we error on missing admins, then it becomes hard to tell if the error is due to a missing footprint or if the admin doesn't exist. Erroring on `set_admin` is fine because regardless of the error, you don't have permission to set the admin.

Reply:
> then it becomes hard to tell if the error is due to a missing footprint or if the admin doesn't exist.

This is generally true for all the host functions and that's something I'd address separately (I don't recall if there is an issue for that, but basically we don't want `try_call` to be able to handle some error kinds gracefully). I don't have a super strong opinion on this, it's just that returning error is more consistent with `set_admin` and also keeps interface simpler (as option would be there to serve exactly one case ever).

Reply:
Hmm ok updated.